orca-front50/src/test/groovy/com/netflix/spinnaker/orca/front50/tasks/SavePipelineTaskSpec.groovy

```diff
@@ -126,4 +126,67 @@ class SavePipelineTaskSpec extends Specification {
     then:
     receivedIndex == newIndex
   }
+
+  def "should update runAsUser with service account in each trigger where it is not set "() {
+    given:
+    String runAsUser
+    def expectedRunAsUser = 'my-pipeline@managed-service-account'
+    def pipeline = [
+      application: 'orca',
+      name: 'my pipeline',
+      stages: [],
+      triggers: [
+        [
+          type: 'cron',
+          enabled: true
+        ]
+      ]
+    ]
+    def stage = new Stage(Execution.newPipeline("orca"), "whatever", [
+      pipeline: Base64.encoder.encodeToString(objectMapper.writeValueAsString(pipeline).bytes)
+    ])
+
+    when:
+    stage.getContext().put('pipeline.serviceAccount', expectedRunAsUser)
+    front50Service.savePipeline(_) >> { Map<String, Object> newPipeline ->
+      runAsUser = newPipeline.get("triggers")?.get(0)?.get('runAsUser')
```

Comment:
In groovy you might as well just walk down the map without `.get`: `newPipeline.triggers[0].runAsUser`.